lib/Commands/NinjaBuildCommand.cpp

```diff
@@ -718,6 +728,191 @@ class BuildContext : public ExecutionQueueDelegate {
       isCancelled = true;
     }
   }
+
+  unsigned getNumPossibleMaxCommands() const {
+    // Compute the "possible" number of maximum commands that will be
+    // run. This is only the "possible" max because we can start running
+    // commands before dependency scanning is complete -- we include the
+    // number of commands that are being scanned so that this number will
+    // always be greater than the number of commands that have been executed
+    // until the very last command is run.
+    int totalPossibleMaxCommands = numCommandsCompleted + numCommandsScanning;
+
+    // Compute the number of max commands to show, subtracting out all the
+    // commands that we avoided running.
+    int possibleMaxCommands = totalPossibleMaxCommands -
+      (numCommandsUpToDate + numCommandsUpdated);
+
+    return possibleMaxCommands;
+  }
+
+  /// Create a resulting byte from the octal components.
+  static bool fromOctalParts(uint8_t& result, char o1, char o2, char o3) {
```

Comment:
We already have `StringRef::getAsInteger`

Reply:
Great, thanks!